bin/job

```diff
@@ -31,29 +32,38 @@ main() {
   fi
 
   [[ "$SRCTYPE" = "disk" ]] && SRC="$(readlink -f $SRC)"
-  [[ "$SRCTYPE" = "libvirt" && -z "$LIBVIRTSRCURI" ]] && die "Please provide a libvirt uri"
+  [[ "$SRCTYPE" = "libvirt" && -z "$SRCURI" ]] && die "Please provide valid uri"
+  [[ "$SRCTYPE" = "libvirt" && "$SRC" =~ ^vpx && -z "$SRCPASS" ]] && die "Please provide password"
 
-  transformVM "$SRCTYPE" "$SRC" "$LIBVIRTSRCURI"
+  transformVM "$SRCTYPE" "$SRC" "$SRCURI" "$SRCPASS"
   createEntities
 }
 
 transformVM() {
   local SRCTYPE=$1
   local SRC="$2"
-  local LIBVIRTSRCURI="$3"
+  local SRCURI="$3"
+  local SRCPASS="$4"
   local WD="$(basename $SRC).d"
 
   echo "  Converting source: $SRC"
   mkdir -p $WD
 
-  ls -shal $SRC
+  ls -shal $WD
 
   local NAME=$(basename $SRC | sed "s/[^a-zA-Z0-9-]/-/g" | tr "[:upper:]" "[:lower:]")
 
   #
   # generate disk and domxml
-  [[ "$LIBVIRTSRCURI" ]] && LIBVIRTSRCURI="-ic $LIBVIRTSRCURI"
-  virt-v2v -i "$SRCTYPE" $LIBVIRTSRCURI "$SRC" -o local -on $NAME -oa sparse -of raw -os $WD --machine-readable
+  [[ "$SRCURI" ]] && SRCURI="-ic $SRCURI"
+  if [[ "SRCPASS" ]] ; then
+    echo $SRCPASS > pass
```

Comment:
Here - instead of creating a file, we could read the file which could be mounted from the Secret object, see https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

Reply:
Sure, will explore.